.claude/hooks/pre-commit-quality-gate.sh

```
@@ -88,9 +95,32 @@ else
   echo "  ⏭️  No ESLint config detected — skipping"
 fi
 
-# ── Step 3: CodeRabbit Review (non-blocking) ────────────────────────────────
+# ── Step 3: Run Tests ───────────────────────────────────────────────────────
+echo ""
+echo "🧪 [3/4] Running tests..."
+
+# Check if test script exists in package.json using jq
+if [ -f "package.json" ]; then
+  HAS_TEST_SCRIPT=$(echo "$INPUT" | jq -r 'if input.scripts.test then "true" else "false" end' < package.json 2>/dev/null || echo "false")
```

Shellcheck (0.11.0)

[error] This redirection overrides piped input. To use both, merge or pass filenames.

(SC2259)